core/src/test/java/io/github/tcdl/msb/api/ChannelMonitorIT.java

```diff
@@ -37,10 +49,143 @@ public void testAnnouncement() throws InterruptedException {
             announcementReceived.countDown();
         };
 
-        DefaultChannelMonitorAggregator channelMonitorAggregator = new DefaultChannelMonitorAggregator(msbContext, new ScheduledThreadPoolExecutor(1), handler);
+        ChannelMonitorAggregator channelMonitorAggregator = msbContext.getObjectFactory().createChannelMonitorAggregator(handler);
         channelMonitorAggregator.start(false, ChannelMonitorAggregator.DEFAULT_HEARTBEAT_INTERVAL_MS, ChannelMonitorAggregator.DEFAULT_HEARTBEAT_TIMEOUT_MS);
 
         channelMonitorAgent.producerTopicCreated(TOPIC_NAME);
         assertTrue("Announcement was not received", announcementReceived.await(RequesterResponderIT.MESSAGE_TRANSMISSION_TIME, TimeUnit.MILLISECONDS));
     }
+
+    @Test
+    public void testAnnouncementUnexpectedMessage() throws InterruptedException {
+        String TOPIC_NAME = "topic1";
+        ChannelMonitorAgent channelMonitorAgent = new DefaultChannelMonitorAgent(msbContext);
+
+        CountDownLatch announcementReceived = new CountDownLatch(1);
+        Callback<AggregatorStats> handler = stats -> {
+            assertTrue(stats.getTopicInfoMap().containsKey(TOPIC_NAME));
+            assertEquals(1, stats.getTopicInfoMap().get(TOPIC_NAME).getProducers().size());
+            announcementReceived.countDown();
+        };
+
+        //simulate unexpectedMessage was consumed from in TOPIC_ANNOUNCE namespace
+        MockAdapter.pushRequestMessage(Utils.TOPIC_ANNOUNCE,
+                Utils.toJson(TestUtils.createMsbRequestMessageWithSimplePayload(Utils.TOPIC_ANNOUNCE), msbContext.getPayloadMapper()));
+
+        ChannelMonitorAggregator channelMonitorAggregator = msbContext.getObjectFactory().createChannelMonitorAggregator(handler);
+        channelMonitorAggregator.start(false, ChannelMonitorAggregator.DEFAULT_HEARTBEAT_INTERVAL_MS, ChannelMonitorAggregator.DEFAULT_HEARTBEAT_TIMEOUT_MS);
+        channelMonitorAgent.producerTopicCreated(TOPIC_NAME);
+
+        assertTrue("Announcement was not received", announcementReceived.await(RequesterResponderIT.MESSAGE_TRANSMISSION_TIME * 2, TimeUnit.MILLISECONDS));
+    }
+
+    @Test
+    public void testHeartbeatMessage() throws InterruptedException {
+        String TOPIC_NAME = "topic2";
+
+        Map<String, AgentTopicStats> topicInfoMap = new HashMap<>();
+        Instant lastProduced = Instant.now().minusMillis(20000);
```

Comment:
Looks dangerous. I would suggest using fixed clock here to avoid problems with slow Travis. In fact you could use constants here, why bother with subtractions?